src/core_plugins/input_control_vis/public/control/list_control_factory.js

```diff
@@ -89,7 +95,12 @@ class ListControl extends Control {
 }
 
 export async function listControlFactory(controlParams, kbnApi, useTimeFilter) {
-  const indexPattern = await kbnApi.indexPatterns.get(controlParams.indexPattern);
+  let indexPattern;
+  try {
+    indexPattern = await kbnApi.indexPatterns.get(controlParams.indexPattern);
+  } catch (err) {
+    // ignore not found error and return control so it can be displayed in disabled state.
+  }
```

Comment:
Third time I've seen this kind of code. Two times is a "maybe I should consolodate". Three + is generally a "Yep. probably a good idea."

Reply:
I am not so sure about adding  a function to encapsulate the try/catch. It will just add another layer of abstraction that has to be dug into to figure out how things work. Also, maybe the callers will want to handle failure differently in the future.